rcl/test/rcl/test_security_directory.cpp

```diff
@@ -49,33 +48,40 @@ static int unsetenv_wrapper(const char * var_name)
 class TestGetSecureRoot : public ::testing::Test
 {
 protected:
+  rcl_allocator_t allocator;
+  char * root_path;
+
   void SetUp() override
   {
+    // Reset rcutil error global state in case a previously
+    // running test has failed.
+    rcl_reset_error();
+
     // Always make sure the variable we set is unset at the beginning of a test
     unsetenv_wrapper(ROS_SECURITY_ROOT_DIRECTORY_VAR_NAME);
     unsetenv_wrapper(ROS_SECURITY_NODE_DIRECTORY_VAR_NAME);
     unsetenv_wrapper(ROS_SECURITY_LOOKUP_TYPE_VAR_NAME);
+    allocator = rcl_get_default_allocator();
+    root_path = "";
```

Comment:
If we keep using a pointer, setting it to "" is dangerous because we should not free the memory in this case.
Using nullptr would be preferable.

Reply:
done.